script-test/playbackstrategies/msestrategytest.js

```diff
@@ -1,491 +1,579 @@
-        mockDashMediaPlayer.create.and.returnValue(mockDashInstance);
+          mockDashjs.MediaPlayer.and.returnValue(mockDashMediaPlayer);
+          mockDashMediaPlayer.create.and.returnValue(mockDashInstance);
 
-        mockVideoElement.addEventListener.and.callFake(function (eventType, handler) {
-          eventHandlers[eventType] = handler;
+          mockVideoElement.addEventListener.and.callFake(function (eventType, handler) {
+            eventHandlers[eventType] = handler;
 
-          eventCallbacks = function (event) {
-            eventHandlers[event].call(event);
-          };
-        });
+            eventCallbacks = function (event) {
+              eventHandlers[event].call(event);
+            };
+          });
+
+          // For DVRInfo Based Seekable Range
+          mockDashInstance.isReady.and.returnValue(true);
+          mockDashInstance.getDashMetrics.and.returnValue({
+            getCurrentDVRInfo: function () {
+              return {range: {
+                start: 0,
+                end: 101
+              }};
+            }
+          });
+
+          window.dashjs = mockDashjs;
 
-        // For DVRInfo Based Seekable Range
-        mockDashInstance.isReady.and.returnValue(true);
-        mockDashInstance.getDashMetrics.and.returnValue({
-          getCurrentDVRInfo: function () {
-            return {range: {
-              start: 0,
-              end: 101
-            }};
-          }
+          spyOn(document, 'createElement').and.returnValue(mockVideoElement);
+          mockDashInstance.getDebug.and.returnValue(mockDashDebug);
+        }
+
+        afterEach(function () {
+          mockVideoElement.currentTime = 0;
+          document.body.removeChild(videoContainer);
         });
 
-        window.dashjs = mockDashjs;
+        describe('Transitions', function () {
+          it('canBePaused() Transition is true', function () {
+            setUpMSE();
 
-        spyOn(document, 'createElement').and.returnValue(mockVideoElement);
-        mockDashInstance.getDebug.and.returnValue(mockDashDebug);
-      }
+            expect(mseStrategy.transitions.canBePaused()).toBe(true);
+          });
 
-      afterEach(function () {
-        mockVideoElement.currentTime = 0;
-        document.body.removeChild(videoContainer);
-      });
+          it('should create an audio element and add it to the video container', function () {
+            setUpMSE(null, null, MediaKinds.AUDIO);
 
-      describe('Transitions', function () {
-        it('canBePaused() Transition is true', function () {
-          setUpMSE();
+            expect(videoContainer.childElementCount).toBe(0);
 
-          expect(mseStrategy.transitions.canBePaused()).toBe(true);
-        });
+            mseStrategy.load('src', null, 0);
 
-        it('canBeginSeek() Transition is true', function () {
-          setUpMSE();
+            expect(videoContainer.firstChild).toBe(mockVideoElement);
```

Comment:
This test won't fail if we create a video element instead. We need to change the mock to return a mock audio element when`document.createElement` is called with `'audio'`, and assert on that instead.

Reply:
Also, this test needs to move into the `load` describe block. It's currently in `transitions` but is not related to transitions.